### ISSUE TITLE
Plugin scaffold should protect against bad naming ("palm-" prefix, special chars)

### ISSUE THREAD

**Issue by norton120** [palmetto/palm-cli#49]
**Describe the bug**

This is really a feature, but feels like a bug when it's happening - so I'm calling it a bug first so others will be able to easily find the issue when it happens to them. 

*TLDR* new plugin names need to not be prefixed with `palm-` and be valid python class names.
 

I wasn't clear on if my plugin should be named `palm-git` or `git` -even though the docs and help make it pretty clear. So I screwed that up and had to handhttps://github.com/palmetto/palm-cli/discussions-hack a lot of files to get it working. 
This isn't absolutely a bug but it is pretty very un-intuitive I think, and would be a solid fit to encourage contribution.
Closely related, we create classnames from this name value - so we need to enforce that they are valid python. 
you can't have special chars in there etc etc. 

**Your environment**
- `palm` version: 2.1.0
- OS: Ubuntu 18.04
- Docker version: Docker version 20.10.10, build b485636
- Other dependencies & versions: 

**How to reproduce**
use `palm plugin new --name `
think about the name for 10 minutes, decide it should match the folder and call it `palm-whatever`
spend a long time finding and deleting the `palm` prefixes from everything. 

**Expected behavior**
remove `palm-` prefix from passed names, so people can make the mistake and we correct it for them. yell and abort when the formatted name is not valid python.
OR 
always yell when the name is not valid python, and make it clear you don't need the `palm` prefix. 

**Actual behavior**
Lots and lots of bad naming that needs to be manually corrected.

**Comment by jakeberesford-palmetto** [comment]
Agreed, this can be cleaned up pretty easily and is a great first issue!

**Comment by jakeberesford-palmetto** [comment]
This was resolved in #59
